core/src/main/scala/kafka/log/Log.scala

```diff
@@ -222,6 +222,9 @@ class Log(@volatile var dir: File,
   /* last time it was flushed */
   private val lastFlushedTime = new AtomicLong(time.milliseconds)
 
+  // Cache value of parent directory
+  @volatile var parentDir: String = dir.getParent
```

Comment:
Can we make the mutable field private and expose an accessor? Should this be returning a `File` instead of `String`? Are there other places where we call `dir.getParent` or `dir.getParentFile` that could use this?